[FSDP][2/N] Remove `_fsdp_wrapped_module.flat_param`

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #87114 [FSDP][6/N] Remove FPW!
* #87113 [FSDP][5/N] Update `FlatParamHandle` after FPW deprecation
* #87112 [FSDP][4/N] Rework FPW test to not use FPW
* #87086 [FSDP][3/N] Register `flat_param` to wrapped module
* **#86122 [FSDP][2/N] Remove `_fsdp_wrapped_module.flat_param`**

This removes **direct** usages of `_fsdp_wrapped_module.flat_param` with `_handles[0].flat_param`. The preferred way to access the `flat_param` will be through the handle. We may converge to only storing `self._handles` and no longer `self.params` in the future. Right now, `self.params` is always exactly `[handle.flat_param for handle in self._handles]`.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501

Commit:
[FSDP][2/N] Remove `_fsdp_wrapped_module.flat_param`

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Remove `_fsdp_wrapped_module.flat_param`"


This removes **direct** usages of `_fsdp_wrapped_module.flat_param` with `_handles[0].flat_param`. The preferred way to access the `flat_param` will be through the handle. We may converge to only storing `self._handles` and no longer `self.params` in the future. Right now, `self.params` is always exactly `[handle.flat_param for handle in self._handles]`.

[ghstack-poisoned]

zhaojuanmao reviewed (commented):
are we assuming _fsdp_wrapped_module only has one handle for now?

awgu commented:
> are we assuming _fsdp_wrapped_module only has one handle for now?

Yup.

awgu commented:
@pytorchbot merge

Commit:
Update on "[FSDP][2/N] Remove `_fsdp_wrapped_module.flat_param`"


This removes **direct** usages of `_fsdp_wrapped_module.flat_param` with `_handles[0].flat_param`. The preferred way to access the `flat_param` will be through the handle. We may converge to only storing `self._handles` and no longer `self.params` in the future. Right now, `self.params` is always exactly `[handle.flat_param for handle in self._handles]`.

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Remove `_fsdp_wrapped_module.flat_param`"


This removes **direct** usages of `_fsdp_wrapped_module.flat_param` with `_handles[0].flat_param`. The preferred way to access the `flat_param` will be through the handle. We may converge to only storing `self._handles` and no longer `self.params` in the future. Right now, `self.params` is always exactly `[handle.flat_param for handle in self._handles]`.

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Remove `_fsdp_wrapped_module.flat_param`"


This removes **direct** usages of `_fsdp_wrapped_module.flat_param` with `_handles[0].flat_param`. The preferred way to access the `flat_param` will be through the handle. We may converge to only storing `self._handles` and no longer `self.params` in the future. Right now, `self.params` is always exactly `[handle.flat_param for handle in self._handles]`.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501

[ghstack-poisoned]
